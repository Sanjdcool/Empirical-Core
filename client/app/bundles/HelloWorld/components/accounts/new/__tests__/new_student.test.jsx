import React from 'react';
import { shallow } from 'enzyme';

import NewStudent from '../new_student';

describe('NewStudent component', () => {

  it('renders a signup form', () => {
    const wrapper = shallow(
        <NewStudent signUp={() => null} textInputGenerator={{generate: () => null}} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('applies textInputGenerator generate function to form fields', () => {

    const formFields = [
      {
        name: 'first_name',
        label: 'First Name',
        errorLabel: 'Name'
      },
      {
        name: 'last_name',
        label: 'Last Name',
        errorLabel: 'Name'
      },
      {
        name: 'username'
      },
      {
        name: 'email',
        label: 'Email (optional)',
        errorLabel: 'Email'
      },
      {
        name: 'password'
      }
    ];

    const mockGenerator = jest.fn();
    const wrapper = shallow(
        <NewStudent signUp={() => null} textInputGenerator={{generate: mockGenerator}} />
    );
    expect(mockGenerator.mock.calls[0][0]).toEqual(formFields);
  });

});
