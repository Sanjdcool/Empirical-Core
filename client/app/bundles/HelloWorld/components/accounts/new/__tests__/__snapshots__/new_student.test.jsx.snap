// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewStudent component renders a signup form 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <div
        className="text-center"
    >
        <div>
            <div>
                <div>
                    <h3
                        className="sign-up-header"
                    >
                        Sign up for a Student Account
                    </h3>
                </div>
            </div>
            <auth_sign_up />
        </div>
        <p
            className="text-center support-p"
        >
            We now support Google Classroom!
        </p>
    </div>
    <div
        className="row"
    >
        <div
            className="col-xs-offset-3 col-xs-9"
        >
            <div
                className="col-xs-8 need-a-border"
            />
            <div
                className="row"
            >
                <div
                    className="col-xs-12"
                />
            </div>
            <div
                className="row"
            >
                <div
                    className="col-xs-12"
                >
                    <button
                        className="button-green col-xs-8"
                        id="sign_up"
                        onClick={[Function]}
                    >
                        Sign Up
                    </button>
                </div>
            </div>
            <div
                className="row"
            >
                <div
                    className="col-xs-8"
                >
                    <div
                        className="text-align-center"
                    >
                        By signing up, you agree to our
                        <a
                            href="/tos"
                            target="_blank"
                        >
                             terms of service 
                        </a>
                         and 
                        <a
                            href="/privacy"
                            target="_blank"
                        >
                             privacy policy
                        </a>
                        .
                    </div>
                </div>
            </div>
        </div>
    </div>
</div>,
  "nodes": Array [
    <div>
      <div
            className="text-center"
      >
            <div>
                  <div>
                        <div>
                              <h3
                                    className="sign-up-header"
                              >
                                    Sign up for a Student Account
                              </h3>
                        </div>
                  </div>
                  <auth_sign_up />
            </div>
            <p
                  className="text-center support-p"
            >
                  We now support Google Classroom!
            </p>
      </div>
      <div
            className="row"
      >
            <div
                  className="col-xs-offset-3 col-xs-9"
            >
                  <div
                        className="col-xs-8 need-a-border"
                  />
                  <div
                        className="row"
                  >
                        <div
                              className="col-xs-12"
                        />
                  </div>
                  <div
                        className="row"
                  >
                        <div
                              className="col-xs-12"
                        >
                              <button
                                    className="button-green col-xs-8"
                                    id="sign_up"
                                    onClick={[Function]}
                              >
                                    Sign Up
                              </button>
                        </div>
                  </div>
                  <div
                        className="row"
                  >
                        <div
                              className="col-xs-8"
                        >
                              <div
                                    className="text-align-center"
                              >
                                    By signing up, you agree to our
                                    <a
                                          href="/tos"
                                          target="_blank"
                                    >
                                           terms of service 
                                    </a>
                                     and 
                                    <a
                                          href="/privacy"
                                          target="_blank"
                                    >
                                           privacy policy
                                    </a>
                                    .
                              </div>
                        </div>
                  </div>
            </div>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <new_student
        signUp={[Function]}
        textInputGenerator={
                Object {
                        "generate": [Function],
                      }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Constructor {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "signUp": [Function],
          "textInputGenerator": Object {
            "generate": [Function],
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <div
                    className="text-center"
          >
                    <div>
                              <div>
                                        <div>
                                                  <h3
                                                            className="sign-up-header"
                                                  >
                                                            Sign up for a Student Account
                                                  </h3>
                                        </div>
                              </div>
                              <auth_sign_up />
                    </div>
                    <p
                              className="text-center support-p"
                    >
                              We now support Google Classroom!
                    </p>
          </div>
          <div
                    className="row"
          >
                    <div
                              className="col-xs-offset-3 col-xs-9"
                    >
                              <div
                                        className="col-xs-8 need-a-border"
                              />
                              <div
                                        className="row"
                              >
                                        <div
                                                  className="col-xs-12"
                                        />
                              </div>
                              <div
                                        className="row"
                              >
                                        <div
                                                  className="col-xs-12"
                                        >
                                                  <button
                                                            className="button-green col-xs-8"
                                                            id="sign_up"
                                                            onClick={[Function]}
                                                  >
                                                            Sign Up
                                                  </button>
                                        </div>
                              </div>
                              <div
                                        className="row"
                              >
                                        <div
                                                  className="col-xs-8"
                                        >
                                                  <div
                                                            className="text-align-center"
                                                  >
                                                            By signing up, you agree to our
                                                            <a
                                                                      href="/tos"
                                                                      target="_blank"
                                                            >
                                                                       terms of service 
                                                            </a>
                                                             and 
                                                            <a
                                                                      href="/privacy"
                                                                      target="_blank"
                                                            >
                                                                       privacy policy
                                                            </a>
                                                            .
                                                  </div>
                                        </div>
                              </div>
                    </div>
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <div
                    className="text-center"
          >
                    <div>
                              <div>
                                        <div>
                                                  <h3
                                                            className="sign-up-header"
                                                  >
                                                            Sign up for a Student Account
                                                  </h3>
                                        </div>
                              </div>
                              <auth_sign_up />
                    </div>
                    <p
                              className="text-center support-p"
                    >
                              We now support Google Classroom!
                    </p>
          </div>
          <div
                    className="row"
          >
                    <div
                              className="col-xs-offset-3 col-xs-9"
                    >
                              <div
                                        className="col-xs-8 need-a-border"
                              />
                              <div
                                        className="row"
                              >
                                        <div
                                                  className="col-xs-12"
                                        />
                              </div>
                              <div
                                        className="row"
                              >
                                        <div
                                                  className="col-xs-12"
                                        >
                                                  <button
                                                            className="button-green col-xs-8"
                                                            id="sign_up"
                                                            onClick={[Function]}
                                                  >
                                                            Sign Up
                                                  </button>
                                        </div>
                              </div>
                              <div
                                        className="row"
                              >
                                        <div
                                                  className="col-xs-8"
                                        >
                                                  <div
                                                            className="text-align-center"
                                                  >
                                                            By signing up, you agree to our
                                                            <a
                                                                      href="/tos"
                                                                      target="_blank"
                                                            >
                                                                       terms of service 
                                                            </a>
                                                             and 
                                                            <a
                                                                      href="/privacy"
                                                                      target="_blank"
                                                            >
                                                                       privacy policy
                                                            </a>
                                                            .
                                                  </div>
                                        </div>
                              </div>
                    </div>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <new_student
    signUp={[Function]}
    textInputGenerator={
        Object {
            "generate": [Function],
          }
    }
/>,
}
`;
